nixos/tests/systemd-confinement.nix

```diff
@@ -48,16 +48,22 @@ import ./make-test-python.nix {
       { config.confinement.mode = "chroot-only";
         testScript = ''
           with subtest("chroot-only confinement"):
-              paths = machine.succeed('chroot-exec ls -1 / | paste -sd,').strip()
-              assert_eq(paths, "bin,nix,run")
+              paths = machine.succeed('chroot-exec ls -Am /').strip()
+              assert_eq(paths, "bin, dev, etc, nix, proc, root, run, sys, usr, var")
+              # systemd’s setup_namespace() calls base_filesystem_create()
+              # which creates those usual top level directories.
+              # Check that they're empty.
+              paths = machine.succeed('chroot-exec ls -Am /dev /etc /proc /root /sys /usr /var | paste -sd,').strip()
```

Comment:
(nitpick: if you need to change something at some point, find's output might be more stable e.g. just the list of directories)

Reply:
I tried `find` but it was not available in the chroot and it was not immediately obvious to me how to put it in there, so I sticked with `ls`. Isn't `ls` output stable/sorted enough here?

Reply:
I'd have expected it to be pulled in the root if you use it with `${pkgs.findutils}/bin/find`, but if you tried I guess this is good enough (I'm not really that worried ls output would change -- it's more of a "it's not really easy to parse with multiple directories content")